From fe9f379274b416cf0cb6cc32e7feb45d79c79b13 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Wed, 7 Dec 2016 23:39:55 +0000
Subject: [PATCH] Force Solus compatibility

We rebuild mesa as emul32 and native, and llvm-config emits invalid
library directories and CFLAGS. We strip -m64 and explicitly set
the library directory to use multilib paths.

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 configure.ac | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/configure.ac b/configure.ac
index adca49d31e..7cb017abb4 100644
--- a/configure.ac
+++ b/configure.ac
@@ -964,6 +964,7 @@ strip_unwanted_llvm_flags() {
         -e 's/-fomit-frame-pointer\>//g' \
         -e 's/-fvisibility-inlines-hidden\>//g' \
         -e 's/-fPIC\>//g' \
+        -e 's/-m64\>//g' \
         -e 's/-fstack-protector-strong\>//g'
 }
 
@@ -978,6 +979,17 @@ llvm_set_environment_variables() {
         LLVM_INCLUDEDIR=`$LLVM_CONFIG --includedir`
         LLVM_LIBDIR=`$LLVM_CONFIG --libdir`
 
+        case "$host_cpu" in
+            i*86)
+                LLVM_LIBDIR="/usr/lib32"
+                LLVM_LDFLAGS="-L/usr/lib32"
+                ;;
+            *)
+                LLVM_LIBDIR="/usr/lib64"
+                LLVM_LDFLAGS="-L/usr/lib64"
+                ;;
+        esac
+
         AC_COMPUTE_INT([LLVM_VERSION_MAJOR], [LLVM_VERSION_MAJOR],
             [#include "${LLVM_INCLUDEDIR}/llvm/Config/llvm-config.h"])
         AC_COMPUTE_INT([LLVM_VERSION_MINOR], [LLVM_VERSION_MINOR],
-- 
2.11.0

